waspc/src/Wasp/Generator/DbGenerator/Jobs.hs

```diff
@@ -130,56 +141,48 @@ seed projectDir seedName =
+        ["db", "execute", "--stdin", "--schema", SP.fromAbsFile absSchemaPath]
 
 -- | Runs `prisma studio` - Prisma's db inspector.
 runStudio :: Path' Abs (Dir ProjectRootDir) -> J.Job
-runStudio projectDir = runPrismaCommandAsDbJob projectDir $ \schema ->
-  ["studio", "--schema", SP.fromAbsFile schema]
-
-generatePrismaClient :: Path' Abs (Dir ProjectRootDir) -> (String, String) -> JobType -> J.Job
-generatePrismaClient projectDir prismaClientOutputDirEnv jobType =
-  runPrismaCommandAsJobWithExtraEnv jobType envVars projectDir $ \schema ->
-    ["generate", "--schema", SP.fromAbsFile schema]
+runStudio projectRootDir =
+  runPrismaCommandAsJob projectRootDir ["studio", "--schema", SP.fromAbsFile schema]
   where
-    envVars = [prismaClientOutputDirEnv]
+    schema = projectRootDir </> dbSchemaFileInProjectRootDir
 
-runPrismaCommandAsDbJob ::
-  Path' Abs (Dir ProjectRootDir) ->
-  (Path' Abs (File PrismaDbSchema) -> [String]) ->
-  J.Job
-runPrismaCommandAsDbJob projectDir makeCmdArgs =
-  runPrismaCommandAsJob J.Db projectDir makeCmdArgs
+generatePrismaClient :: Path' Abs (Dir ProjectRootDir) -> J.Job
+generatePrismaClient projectRootDir =
+  runPrismaCommandAsJob projectRootDir cmdArgs
+  where
+    cmdArgs =
+      [ "generate",
+        "--schema",
+        SP.fromAbsFile $ projectRootDir </> dbSchemaFileInProjectRootDir
+      ]
 
-runPrismaCommandAsJob ::
-  JobType ->
-  Path' Abs (Dir ProjectRootDir) ->
-  (Path' Abs (File PrismaDbSchema) -> [String]) ->
-  J.Job
-runPrismaCommandAsJob jobType projectDir makeCmdArgs =
-  runPrismaCommandAsJobWithExtraEnv jobType [] projectDir makeCmdArgs
+runPrismaCommandAsJob :: Path' Abs (Dir ProjectRootDir) -> [String] -> J.Job
+runPrismaCommandAsJob projectRootDir cmdArgs =
+  runPrismaCommandAsJobWithExtraEnv [] projectRootDir cmdArgs
 
 runPrismaCommandAsJobWithExtraEnv ::
-  JobType ->
   [(String, String)] ->
   Path' Abs (Dir ProjectRootDir) ->
-  (Path' Abs (File PrismaDbSchema) -> [String]) ->
+  [String] ->
   J.Job
-runPrismaCommandAsJobWithExtraEnv jobType envVars projectDir makeCmdArgs =
-  runNodeCommandAsJobWithExtraEnv envVars serverDir (absPrismaExecutableFp projectDir) (makeCmdArgs schemaFile) jobType
+runPrismaCommandAsJobWithExtraEnv envVars projectRootDir cmdArgs =
+  runNodeCommandAsJobWithExtraEnv envVars waspProjectDir (absPrismaExecutableFp waspProjectDir) cmdArgs J.Db
   where
-    serverDir = projectDir </> serverRootDirInProjectRootDir
-    schemaFile = projectDir </> dbSchemaFileInProjectRootDir
+    waspProjectDir = projectRootDir </> waspProjectDirFromProjectRootDir
 
 -- | NOTE: The expectation is that `npm install` was already executed
 -- such that we can use the locally installed package.
 -- This assumption is ok since it happens during compilation now.
-absPrismaExecutableFp :: Path' Abs (Dir ProjectRootDir) -> FilePath
-absPrismaExecutableFp projectDir = SP.fromAbsFile prismaExecutableAbs
+absPrismaExecutableFp :: Path' Abs (Dir WaspProjectDir) -> FilePath
```

Comment:
Aha, so this basically says that Prisma is now living in user node_modules, right?

Reply:
Exactly!

Reply:
Should we mention here why we expect that prisma is installed or user's node_modules, or that we expect it? I am maybe a bit out of context here so disregard this if it doesn't make sense.

Reply:
We're specifying it as a dependency, and the comment above says it relies on "npm install" being done, so I think it's fine. I can remove the "This assumption is ok..." part.

Reply:
I am out of context here, so whatever you think is best. For stuff like this, extra comments don't hurt.

Reply:
We can comment something like "This command assumes the project's top level `npm install` was already executed and that `prisma` exists inside the top-level `node_modules` .